Kconfig.zephyr

```diff
@@ -167,6 +167,13 @@ config LINKER_SORT_BY_ALIGNMENT
 	  in decreasing size of symbols. This helps to minimize
 	  padding between symbols.
 
+config SRAM_VECTOR_TABLE
+	bool "Place the vector table in SRAM instead of flash"
+	default y if SOC_SERIES_STM32F0X
```

Comment:
Is this option too Cortex-M specific? Do we anticipate that we would need this for other ARCHes? If not, consider defining the option in the common ARM SoC Kconfig, instead.

In any case, it is cleaner if the main Kconfig.zephyr file does not contain (more) SoC- specific defines. Therefore, I would just define SRAM_VECTOR_TABLE here (without the default y setting for STM32), and re-define the option in the  STM32F0X Kconfig.defconfig as default y.

Reply:
@erwango thoughts?

Reply:
@ioannisg I'm fine with your proposed approach that seems cleaner indeed.

Reply:
Am I moving this to ARM Kconfig, or changing the default in the STM32F0 defconfig?

Reply:
If we only anticipate Cortex-M usage of the symbol, I would do (1), and then (2) is up to you. 
If not, just Do (2) :)

Reply:
Default has been moved to STM32F0 defconfig